src/classes/NoaaRestClient.php

```diff
@@ -0,0 +1,297 @@
+<?php
+/**
+ * NoaaRestClient - A client for accessing NOAA's REST API
+ * 
+ * This class replaces the legacy SOAP client for accessing NOAA's weather data.
+ * It handles all API communication, response parsing, and data transformation to match
+ * the format expected by the existing application.
+ */
+class NoaaRestClient {
+    private $baseUrl = 'https://api.weather.gov';
+    private $userAgent = 'TakeASweaterApp/1.0 (https://takeasweater.com, contact@takeasweater.com)';
+    private $cacheDir = '/tmp/takeasweater_cache';
+    private $cacheTime = 3600; // 1 hour cache for location metadata
+    
+    /**
+     * Constructor - creates cache directory if it doesn't exist
+     */
+    public function __construct() {
+        if (!file_exists($this->cacheDir)) {
+            mkdir($this->cacheDir, 0755, true);
+        }
+    }
+    
+    /**
+     * Get high and low temperatures for the given location and date range
+     * 
+     * @param string $startDate The start date in YYYY-MM-DD format
+     * @param int $numDays Number of days to forecast
+     * @param string $unit Unit system (e for English, m for Metric)
+     * @param string $format Format of the data (not used in REST API)
+     * @param float $lat Latitude
+     * @param float $lon Longitude
+     * @return array|false Formatted weather data array or false on failure
+     */
+    public function get_highs_lows($startDate, $numDays, $unit, $format, $lat, $lon) {
+        try {
+            // Step 1: Get grid point information for this location
+            $gridInfo = $this->getPointMetadata($lat, $lon);
+            if (!$gridInfo) {
+                error_log("Failed to get grid information for location ($lat, $lon)");
+                return false;
+            }
+            
+            // Step 2: Get the forecast data using the grid information
+            $forecastData = $this->getForecast($gridInfo['gridId'], $gridInfo['gridX'], $gridInfo['gridY']);
+            if (!$forecastData) {
+                error_log("Failed to get forecast data for grid: {$gridInfo['gridId']}/{$gridInfo['gridX']},{$gridInfo['gridY']}");
+                return false;
+            }
+            
+            /////////////////// DEBUG ///////////////////
+            // Add debug output to see raw API data
+            file_put_contents('/tmp/api_raw_data.json', json_encode($forecastData, JSON_PRETTY_PRINT));
```

Comment:
Same comment as before about adding a way to enable/disable this kind of debugging output. It should be disabled by default.